src/coupon/coupon-selector.ts

```diff
@@ -13,11 +14,11 @@ export default class CouponSelector {
         return this._coupon.data;
     }
 
-    getRemoveError(): Error | undefined {
+    getRemoveError(): RequestError | undefined {
```

Comment:
Do we need to specify the error response type?

Reply:
Actually I think we might not want to. The generic type is the body of the original error response, but part of the work I did was to standardise how we return RequestErrors from the 3 different APIs that we interact with (internal, payments, storefront), so we shouldn't really need to access the original API payload (even though it's available)